.github/workflows/pgroll-spec.yml

```diff
@@ -0,0 +1,41 @@
+name: Update pgroll spec
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */3 * * 1-5' # once every 3 hours on week days
+
+jobs:
+  update-deps:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Read .nvmrc
+        run: echo "NVMRC=$(cat .nvmrc)" >> $GITHUB_OUTPUT
+        id: nvm
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+
+      - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ steps.nvm.outputs.NVMRC }}
+          cache: 'pnpm'
+
+      - name: Check for pgroll spec updates
+        run: cd packages/pgroll && pnpm generate
```

Comment:
I'm late, but does this step fail if there are no changes generated? 

Otherwise how does the action avoid creating a PR when nothing has changed?

Reply:
The step from `peter-evans/create-pull-request@v4` avoids creating a PR if there are no changes on commit.

There's a couple of things I want to improve though:

- Run prettier before so that the action doesn't fail when the precommit hook removes the formatting differences
- Change the branch name so that the action updates the existing open PR